internal/services/netapp/netapp_account_encryption_resource.go

```diff
@@ -294,17 +321,52 @@ func expandEncryption(ctx context.Context, input string, keyVaultsClient *keyVau
 
 	keyId, err := keyVaultParse.ParseOptionallyVersionedNestedKeyID(input)
 	if err != nil {
-		return nil, fmt.Errorf("parsing `key_vault_key_id`: %+v", err)
+		return nil, fmt.Errorf("parsing `key_vault_key_id` %q: %+v", input, err)
 	}
 
-	keyVaultID, err := keyVaultsClient.KeyVaultIDFromBaseUrl(ctx, subscriptionID, keyId.KeyVaultBaseUrl)
-	if err != nil {
-		return nil, fmt.Errorf("retrieving the resource id the key vault at url %q: %s", keyId.KeyVaultBaseUrl, err)
-	}
+	var keyVaultResourceID string
+
+	// For cross-tenant scenarios, we can't lookup the key vault from the current tenant
+	// Instead, we need to construct the resource ID manually or use a provided resource ID
+	if model.FederatedClientID != "" {
+		// Cross-tenant scenario: try to construct the key vault resource ID from the URL
+		// Extract the key vault name from the URL for all Azure cloud environments
+		keyVaultName := ""
+		if keyId.KeyVaultBaseUrl != "" {
+			const httpsPrefix = "https://"
+			if strings.HasPrefix(keyId.KeyVaultBaseUrl, httpsPrefix) {
+				// Remove the "https://" prefix
+				urlWithoutProtocol := keyId.KeyVaultBaseUrl[len(httpsPrefix):]
+
+				// Find the first dot to separate the vault name from the domain
+				dotIndex := strings.Index(urlWithoutProtocol, ".")
+				if dotIndex > 0 {
+					keyVaultName = urlWithoutProtocol[:dotIndex]
+				}
+			}
+		}
 
-	parsedKeyVaultID, err := commonids.ParseKeyVaultID(pointer.From(keyVaultID))
-	if err != nil {
-		return nil, err
+		if keyVaultName == "" {
+			return nil, fmt.Errorf("unable to extract key vault name from URL %q for cross-tenant scenario", keyId.KeyVaultBaseUrl)
+		}
+
+		// For cross-tenant scenarios, use the provided resource ID if available
+		// Otherwise, construct a placeholder resource ID (for backwards compatibility)
+		if model.CrossTenantKeyVaultResourceID != "" {
```

Comment:
It looks like we could get into a situation where `FederatedClientID` is specified but not  `CrossTenantKeyVaultResourceID` and so we won't pass this check to set `keyVaultResourceID`. Should we link the two in the schema through `RequiredWith`?